pkg/core/e2e/e2e_test.go

```diff
@@ -573,6 +573,136 @@ var _ = Describe("Cyphernetes E2E", func() {
 			By("Cleaning up")
 			Expect(k8sClient.Delete(ctx, testDeployment)).Should(Succeed())
 		})
+
+		It("Should execute pattern matching in WHERE clauses correctly", func() {
+			By("Creating test resources")
+			// Create a deployment that owns a replicaset that owns pods
+			testDeployment := &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-deployment-pattern",
+					Namespace: testNamespace,
+					Labels: map[string]string{
+						"app": "pattern-test",
+					},
+				},
+				Spec: appsv1.DeploymentSpec{
+					Replicas: ptr.To(int32(0)),
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"app": "pattern-test",
+						},
+					},
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"app": "pattern-test",
+							},
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "nginx",
+									Image: "nginx:latest",
+								},
+							},
+						},
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, testDeployment)).Should(Succeed())
+
+			// Wait for deployment and its resources to be ready
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{
+					Namespace: testNamespace,
+					Name:      "test-deployment-pattern",
+				}, &appsv1.Deployment{})
+			}, timeout, interval).Should(Succeed())
+
+			By("Executing a pattern matching query")
+			provider, err := apiserver.NewAPIServerProvider()
+			Expect(err).NotTo(HaveOccurred())
+
+			executor, err := core.NewQueryExecutor(provider)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Query to find pods owned by the deployment through a replicaset
```

Comment:
nit - I think it should be:
```suggestion
			// Query to find pods not owned by the deployment through a replicaset
```

Reply:
Thanks good catch, improved this comment and the one that followed